charts/hivemq-platform-operator/templates/deployment.yml

```diff
@@ -51,27 +51,31 @@ spec:
       containers:
         - env:
             - name: JAVA_OPTS
-              value: {{.Values.javaOpts | quote }}
-            {{- if and (ne .Values.selector "nil") (.Values.selector) }}
+              value: {{ .Values.javaOpts | quote }}
+            {{- with .Values.selector }}
```

Comment:
I assume `with` will ignore `nil` and `""`? We need to support both, since users still might have a custom `values.yaml` with `nil` as value.

Reply:
Previous default values will get overwritten with the new ones, so customers having default `nil` values will get new default ones instead, so it won't be applicable anymore.

For example, with the following example where the `operator-values.yaml` is something like:
```yaml
resources:
  cpu: 1025m
  memory: 1Gi
```

With:

```
$ helm upgrade --install operator hivemq/hivemq-platform-operator -f ./operator-values.yaml --version 0.2.10 <-- current release version

$ kubectl describe pod hivemq-operator-5bcd6dbf-qk7ld.   <-- Check there is no EnvVar for the `nil` values added (for example selector, or any TLS configuration)                               
...
    Environment:
      JAVA_OPTS:                                                                         -XX:+UnlockExperimentalVMOptions -XX:InitialRAMPercentage=75 -XX:MaxRAMPercentage=75
      KUBERNETES_NAMESPACE:                                                              default (v1:metadata.namespace)
      hivemq.platform.operator.release.name:                                             operator
      hivemq.platform.operator.init.image:                                               docker.io/hivemq/hivemq-platform-operator-init:1.5.2
      hivemq.platform.operator.image.pull.secret:                                        
      hivemq.platform.operator.log.level:                                                INFO
      hivemq.platform.operator.log.configuration:                                        false
      hivemq.platform.operator.serviceaccount.create:                                    true
      hivemq.platform.operator.serviceaccount.validate:                                  true
      hivemq.platform.operator.serviceaccount.name:                                      
      hivemq.platform.operator.serviceaccount.permissions.create:                        true
      hivemq.platform.operator.serviceaccount.permissions.validate:                      true
      hivemq.platform.operator.statefulset.rolling-restart-on-template-metadata-change:  false
      hivemq.platform.operator.quarkus.log.level:                                        INFO
      hivemq.platform.operator.sdk.log.level:                                            INFO
      hivemq.platform.operator.network.log.level:                                        WARN
      quarkus.http.port:                                                                 8080
      quarkus.http.ssl-port:                                                             8443
...      

$ helm get values operator --all                       <-- Get actual computed values. See the `nil` values                                          
COMPUTED VALUES:
env: []
hivemqPlatformServiceAccount:
  create: true
  name: ""
  permissions:
    create: true
    validate: true
  validate: true
hivemqPlatformStatefulSet:
  rollingRestartOnTemplateMetadataChange: false
http:
  port: 8080
https:
  port: 8443
image:
  initImageName: hivemq-platform-operator-init
  name: hivemq-platform-operator
  pullPolicy: IfNotPresent
  pullSecretName: ""
  repository: docker.io/hivemq
  tag: 1.5.2
javaOpts: -XX:+UnlockExperimentalVMOptions -XX:InitialRAMPercentage=75 -XX:MaxRAMPercentage=75
logLevel: INFO
podScheduling:
  affinity: null
  tolerations: null
podSecurityContext:
  enabled: false
  runAsNonRoot: true
rbac:
  create: true
resources:
  cpu: 1025m
  memory: 1Gi
selector: nil
serviceAccount:
  create: true
  name: ""
tls:
  keystorePassword: nil
  secretName: nil
  truststorePassword: nil
```

Then upgrade the already running HiveMQ Platform Operator release using this chart locally with these changes from this PR as below, and you will see that only the new CRD environment variable have been added:
```
helm upgrade operator ./charts/hivemq-platform-operator -f operator-values.yaml
...
$ kubectl describe pod hivemq-operator-5bcd6dbf-qk7ld    <-- Check there is no EnvVar for the `nil` values added (for example selector, or any TLS configuration)          
    Environment:
      JAVA_OPTS:                                                                         -XX:+UnlockExperimentalVMOptions -XX:InitialRAMPercentage=75 -XX:MaxRAMPercentage=75
      KUBERNETES_NAMESPACE:                                                              default (v1:metadata.namespace)
      hivemq.platform.operator.release.name:                                             operator
      hivemq.platform.operator.crd.apply:                                                true
      hivemq.platform.operator.crd.wait-until-ready:                                     true
      hivemq.platform.operator.crd.wait-until-ready.timeout:                             PT10S
      hivemq.platform.operator.init.image:                                               docker.io/hivemq/hivemq-platform-operator-init:1.5.2
      hivemq.platform.operator.image.pull.secret:                                        
      hivemq.platform.operator.log.level:                                                INFO
      hivemq.platform.operator.log.configuration:                                        false
      hivemq.platform.operator.serviceaccount.create:                                    true
      hivemq.platform.operator.serviceaccount.validate:                                  true
      hivemq.platform.operator.serviceaccount.name:                                      
      hivemq.platform.operator.serviceaccount.permissions.create:                        true
      hivemq.platform.operator.serviceaccount.permissions.validate:                      true
      hivemq.platform.operator.statefulset.rolling-restart-on-template-metadata-change:  false
      hivemq.platform.operator.quarkus.log.level:                                        INFO
      hivemq.platform.operator.sdk.log.level:                                            INFO
      hivemq.platform.operator.network.log.level:                                        WARN
      quarkus.http.port:                                                                 8080
      quarkus.http.ssl-port:                                                             8443

$ helm get values operator --all   <-- Get actual computed values. See how the `nil` values look like now                                                       
COMPUTED VALUES:
crd:
  apply: true
  waitReady: true
  waitTimeout: PT10S
env: []
hivemqPlatformServiceAccount:
  create: true
  name: ""
  permissions:
    create: true
    validate: true
  validate: true
hivemqPlatformStatefulSet:
  rollingRestartOnTemplateMetadataChange: false
http:
  port: 8080
https:
  port: 8443
image:
  initImageName: hivemq-platform-operator-init
  name: hivemq-platform-operator
  pullPolicy: IfNotPresent
  pullSecretName: ""
  repository: docker.io/hivemq
  tag: 1.5.2
javaOpts: -XX:+UnlockExperimentalVMOptions -XX:InitialRAMPercentage=75 -XX:MaxRAMPercentage=75
logLevel: INFO
podScheduling:
  affinity: null
  tolerations: null
podSecurityContext:
  enabled: false
  runAsNonRoot: true
rbac:
  create: true
resources:
  cpu: 1025m
  memory: 1Gi
selector: ""
serviceAccount:
  create: true
  name: ""
tls:
  keystorePassword: ""
  secretName: ""
  truststorePassword: ""      
```

Reply:
Damn... you're right. I was using a wrong example for the custom values YAML file. If I use all previous default values as my custom values YAML file, then it fails:

```yaml
...
selector: nil
...
tls:
  secretName: nil
  keystorePassword: nil
  truststorePassword: nil
```

These default values ones are only overwritten when they are not present in the values YAML file used 😞 
Will put them back.